File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        hashMap = {}

        for s in strs:
            sorted_str = ''.join(sorted(s))
            if sorted_str not in hashMap:
                hashMap[sorted_str] = []
            hashMap[sorted_str].append(s)
        
        return list(hashMap.values())

        

File: 0049-group-anagrams/0049-group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
     let hashMap = new Map()

    for(let str of strs){
        const sorted = str.split("").sort().join("")

        if(!hashMap.has(sorted)){
            hashMap.set(sorted,[])
        }
        hashMap.get(sorted).push(str)
    }
    const grouped = Array.from(hashMap.values())
    return grouped
};